osbrain/tests/test_agent_sync_publications.py

```diff
@@ -73,6 +73,164 @@ def on_error(agent):
     agent.error_log.append('error')
 
 
+@pytest.mark.parametrize('socket_type', ['PUB', 'SYNC_PUB'])
+@pytest.mark.parametrize('handler', [append_received, {b'': append_received}])
+def test_unsubscribe(nsproxy, socket_type, handler):
+    '''
+    Test unsubscribing from topics works fine.
+    '''
+    server = run_agent('server')
+    client = run_agent('client')
+
+    addr = server.bind(socket_type, alias='pub', handler=lambda: None)
+    client.set_attr(received=[])
+    client.connect(addr, alias='sub', handler=handler)
+
+    # Stablish a connection
+    server.each(0.1, 'send', 'pub', 'connecting...')
+    assert wait_agent_attr(client, data='connecting...')
+
+    # Clear subscription to default '' topic (do not receive anything)
+    client.unsubscribe('sub', '')
+
+    # Message not received
+    server.send('pub', 'not received')
+    assert not wait_agent_attr(client, data='not received',
+                               timeout=0.5)
+
+
+@pytest.mark.parametrize('socket_type', ['PUB', 'SYNC_PUB'])
+def test_subscribe(nsproxy, socket_type):
+    '''
+    Test subscribing to various topics/handlers works fine.
+    '''
+    server = run_agent('server')
+    client = run_agent('client')
+
+    addr = server.bind(socket_type, alias='pub', handler=lambda: None)
+    client.set_attr(received=[])
+    client.connect(addr, alias='sub', handler=append_received)
+
+    # Stablish a connection
+    server.each(0.1, 'send', 'pub', 'connecting...')
+    assert wait_agent_attr(client, data='connecting...')
+
+    # Subscribe to two topics
+    client.unsubscribe('sub', '')
+    client.subscribe('sub', handlers={'negate': receive_negate,
+                                      'normal': append_received})
+
+    # Only message from subscribed topic received
+    server.send('pub', 1, topic='negate')
+    server.send('pub', 2, topic='normal')
+    server.send('pub', 'not received', topic='topic_C')
+
+    assert wait_agent_attr(client, data=-1)
+    assert wait_agent_attr(client, data=2)
+    assert not wait_agent_attr(client, data=1, timeout=0.5)
+    assert not wait_agent_attr(client, data=-2, timeout=0.5)
+    assert not wait_agent_attr(client, data='not received',
+                               timeout=0.5)
```

Comment:
I think the first two `assert not wait` might be redundant. But if you wanted to keep them, maybe set the timeout to `0.5` only for the first `assert not wait`. Then for the rest set it to `0` to make it a bit faster?